hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/common/HoodieFlinkEngineContext.java

```diff
@@ -102,12 +102,12 @@ public RuntimeContext getRuntimeContext() {
 
   @Override
   public <I, O> List<O> map(List<I> data, SerializableFunction<I, O> func, int parallelism) {
-    return data.stream().parallel().map(throwingMapWrapper(func)).collect(Collectors.toList());
+    return stream(data, parallelism).map(throwingMapWrapper(func)).collect(Collectors.toList());
   }
```

Comment:
What is the default parallelism for `stream().parallel()` ?

Reply:
@danny0405, the default parallelism is `Runtime.getRuntime().availableProcessors()`.

Reply:
Hi @SteNicholas,  correct me if I'm wrong.  Do we just run the map function sequentially when the parallelism is larger than the default parallelism? And why there is an `OutOfMemoryError ` risk when we use  `stream().parallel()` which will submit some future tasks to the default thread pool which may not cost a lot of memory?

Reply:
@stream2000, I don't think only run the map function sequentially. In order to reduce the OOM risk caused by parallelization, all functions should be handled in this way. BTW, for example, when the parallelism is 1000, the `CleanPlanActionExecutor` uses map function and may casue `OutOfMemoryError` for many filegroups in `stream().parallel()`, which stacktrace like above description.

Reply:
When the parallelism is 1000,  will we run it in 1000 parallelism or `Runtime.getRuntime().availableProcessors()`?  If we use just the parallelism as `Runtime.getRuntime().availableProcessors()`,  will it cause `OutOfMemoryError` since it it not actually a large parallelism?

Reply:
@stream2000, the parallelism of `stream().parallel()` is only `Runtime.getRuntime().availableProcessors()`.

Reply:
> @stream2000, the parallelism of `stream().parallel()` is only `Runtime.getRuntime().availableProcessors()`

So actually the parallelism of `availableProcessors` will cause OOM? If I configure the parallelism just as `Runtime.getRuntime().availableProcessors()` we still get OOM right? Correct me if I'm wrong~